perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/RegistrarManagerImpl.java

```diff
@@ -1702,6 +1703,26 @@ public Application approveApplicationInternal(PerunSession sess, int appId) thro
 					}
 				}
 
+				// get all app items
+				List<ApplicationFormItemData> items = getApplicationDataById(registrarSession, app.getId());
```

Comment:
I do not think that the place where this code is used is right. You want to set names only when you are creating new users.
This way you could re-set them when some user is applying to another Vo.

Reply:
Actually, I have discussed this with @licehammer, and for this widget, we want it to set the names even to existing users.